pkg/apis/network/v1/pki_types.go

```diff
@@ -0,0 +1,60 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PKISpec defines the desired state of PKI.
+// +k8s:openapi-gen=true
+type PKISpec struct {
+	// caCert is information about the CA certificate and key.
+	CACert CertSpec `json:"caCert"`
+
+	// caBundleName is the name of the config-map containing the CA bundle.
+	CABundleName string `json:"caBundleName"`
+
+	// serverCert is the server certificate to create.
+	// It will be created with both ServerAuth and ClientAuth
+	ServerCert CertSpec `json:"serverCert"`
```

Comment:
Maybe include the words "key usage" or something to make that clearer.
Also, why both? That seems weird.
Also, where does the certificate get created to?